### ISSUE TITLE
React-json-tree: Add callback when expanding/closing node.

### ISSUE THREAD

**Issue by BrunoFenzl** [reduxjs/redux-devtools#636]
As far as I could see, there is no way of responding to a click on a node in the tree. I need to get the `keypath` property corresponding the node I clicked on. Currently I'm hijacking `valueRender` to get this value, but of course this also get called, at least for the root node when re-rendering the component.

Is there any interest in this feature? I badly need this feature and could merge make a PR if there is any interest.

**Comment by Methuselah96** [comment]
Yeah, that sounds like a valuable feature. Feel free to make a PR.

**Comment by BrunoFenzl** [comment]
Nice, will do.

**Comment by BrunoFenzl** [comment]
Hi @Methuselah96 .

I'm having a hard time trying to get my environment right. It seems I have all sorts of version mismatches and missing deps. Is there a guide on how to contribute and system requirements anywhere? I tried on two machines with node 12.5 and 14.12.
Thanks in advance!

**Comment by Methuselah96** [comment]
Are you using Yarn? The monorepo uses Yarn workspaces, so it won't work with npm.

**Comment by BrunoFenzl** [comment]
Thanks for the fast response and clarifying this. I have yarn installed but thought it would just execute the commands regardless of how I installed dependencies. I use npm on my projects so I installed everything with `npm -i` as usual. I just learned something new.

It would be really nice to have this in the readme.

**Comment by Methuselah96** [comment]
@BrunoFenzl Glad it worked out. Sorry about the lack of developer documentation, still just trying to clean things up. Feel free to ask if you have any other questions.
